cg/meta/demultiplex/validation.py

```diff
@@ -6,24 +6,25 @@
 from cg.exc import FlowCellError
 
 
-def validate_sample_fastq_file(sample_fastq: Path) -> None:
+def is_valid_sample_fastq_file(sample_fastq: Path, sample_id: str) -> bool:
     """
     Validate that the sample fastq file name is formatted as expected.
 
-    Since many different naming conventions are used, these are the only assumptions which can be made:
+    Assumptions:
     1. The sample fastq file name ends with .fastq.gz
     2. The sample fastq file name contains the lane number formatted as _L<lane_number>
-    3. The sample fastq file is located in a directory containing the sample id formatted as Sample_<sample_id>
+    3. The sample id is present in the parent directory name or in the file name.
```

Comment:
@Vince-janv looked and @karlnyr looked at them before. They make sense because either:
1. The fastq file is in the unaligned directory structure, meaning the parent directory has the sample id
2. The fastq file is in the root directory, meaning the fastq file name contains the sample id